feature: add placeholderText theme color for input placeholders (resolves #5088)

ariane-emory opened the pull request:
Resolves #5088.

This PR adds a new, optional `placeholderText` theme color that allows customizing the color of placeholder text in input fields and textareas. There is no change in beheaviour for users who do not choose to use this new theme property: If the `placeholderText` property is not used, the existing behaviour of using `textMuted` is maintained.

Changes:
- Add `placeholderText` to ThemeColors type with fallback to `textMuted`
- Apply `placeholderColor` prop to <input> components in dialog-select
- Apply styled placeholder text to <textarea> components in dialog-prompt and prompt
- Update JSON schemas to include `placeholderText` property
- Maintain backward compatibility - existing themes work unchanged

To use: 
Simply include a line resembling the following to your theme file:
```json
"placeholderText": "#c00"
```

ariane-emory committed:
feat: Add placeholderText theme color for input placeholders

Fixes #5088 by adding a new optional \`placeholderText\` theme color that allows
customizing the color of placeholder text in input fields and textareas.

Changes:
- Add \`placeholderText\` to ThemeColors type with fallback to \`textMuted\`
- Apply \`placeholderColor\` prop to <input> components in dialog-select
- Apply styled placeholder text to <textarea> components in dialog-prompt and prompt
- Update JSON schemas to include \`placeholderText\` property
- Maintain backward compatibility - existing themes work unchanged

The new field follows ariane-emory's suggestion and provides a minimal,
focused solution for theming placeholder text.

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

ariane-emory committed:
Fix TypeScript error: remove cacheKey from FileContents interface usage

ariane-emory committed:
Merge upstream/dev into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

b0o commented:
The main input textarea also needs this treatment:
```patch
diff --git a/packages/opencode/src/cli/cmd/tui/component/prompt/index.tsx b/packages/opencode/src/cli/cmd/tui/component/prompt/index.tsx
index 7d5bbb9f0..1b13264df 100644
--- a/packages/opencode/src/cli/cmd/tui/component/prompt/index.tsx
+++ b/packages/opencode/src/cli/cmd/tui/component/prompt/index.tsx
@@ -656,7 +656,7 @@ export function Prompt(props: PromptProps) {
             flexGrow={1}
           >
             <textarea
-              placeholder={props.sessionID ? undefined : `Ask anything... "${PLACEHOLDERS[store.placeholder]}"`}
+              placeholder={props.sessionID ? undefined : t`${fg(theme.placeholderText)(`Ask anything... "${PLACEHOLDERS[store.placeholder]}"`)}`}
               textColor={theme.text}
               focusedTextColor={theme.text}
               minHeight={1}
```

b0o commented:
Also, do you think it would be better to add a `placeholderColor` prop to `opentui`'s `textarea` and then pass the color directly, like how the other colors are passed? I grepped the opencode codebase and I don't see `fg()` or the `t` tagged template function from `@opentui/core` being used anywhere else.

ariane-emory committed:
tweak: use placeholderColor here as per bOo's request.

ariane-emory committed:
Merge branch 'fix/placeholder-color-5088' of github.com:ariane-emory/opencode into fix/placeholder-color-5088

ariane-emory committed:
chore: resync this file

ariane-emory commented:
@b0o Addressed your first comment, that part should be good now. I'm going to have to spend a little more time thinking and investigating to come up with an answer I can feel confident in to the question in your latter comment and it's already getting very late here, so I'm going to need to sleep on it and get back to you tomorrow.

ariane-emory committed:
Add placeholderColor prop to OpenTUI Textarea component

- Update TextareaOptions interface to include optional placeholderColor field
- Add _placeholderColor private field and getter/setter methods to TextareaRenderable
- Modify constructor to initialize placeholderColor with fallback to #666666
- Update applyPlaceholder method to use placeholderColor instead of hardcoded fg(#666666)
- Update OpenCode components to use new placeholderColor prop instead of t/fg template functions
- Remove unused t, fg imports from dialog-prompt and prompt components

This enables consistent placeholder color theming across all input components using the same prop-based API pattern

ariane-emory committed:
Update OpenCode to use placeholderColor prop from opentui

- Remove t, fg imports from DialogPrompt and Prompt components
- Change textarea components to use placeholderColor prop instead of t/fg template functions
- This leverages the new placeholderColor prop added to opentui Textarea component
- Provides consistent API pattern matching Input component's placeholderColor usage

ariane-emory committed:
Merge branch 'fix/placeholder-color-5088' of github.com:ariane-emory/opencode into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'upstream/dev' into fix/placeholder-color-5088

ariane-emory committed:
Fix TypeScript error: remove placeholderColor property

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory commented:
Corrupted by a bad merge, will fix.

ariane-emory committed:
fix: uncorrupt

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
tidy: whitespace

ariane-emory committed:
Merge branch 'dev' into repair/fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'origin/dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'fix/placeholder-color-5088' of github.com:ariane-emory/opencode into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge remote-tracking branch 'origin/dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'fix/placeholder-color-5088' of github.com:ariane-emory/opencode into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'fix/placeholder-color-5088' of github.com:ariane-emory/opencode into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'fix/placeholder-color-5088' of github.com:ariane-emory/opencode into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'fix/placeholder-color-5088' of github.com:ariane-emory/opencode into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory committed:
Merge branch 'dev' into fix/placeholder-color-5088

ariane-emory commented:
Doesn't seem to be a whole lot of interest in this one. I'll close it for now, just let me know if you'd like it reopened... I'll keep the branch around for a few more days before deleting it.
